build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+else
+	echo "Exiting: OS should be debian or ubuntu. Please provide the right one"
+	exit
+fi
+
+mkdir ${os}-${flavor}-Glusterfs-${version}
+
+cd ${os}-${flavor}-Glusterfs-${version}
+
+mkdir build packages
+
+echo "Building glusterfs-${version}-${release} for ${flavor}"
+
+cd build
+
+TGZS=(`ls ~/glusterfs-${version}-?-*/build/glusterfs-${version}.tar.gz`)
+echo ${TGZS[0]}
+
+if [ -z ${TGZS[0]} ]; then
+        echo "wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz"
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
+
+cd glusterfs-debian/
+
+git checkout -b ${flavor}-${series}-local origin/${flavor}-glusterfs-${series}
+
+if [ "$os" == "ubuntu" ]; then
+	sed -i "1i glusterfs (${version}-${os}1~${flavor}1) ${flavor}; urgency=medium\n\n * GlusterFS ${version} GA\n\n -- GlusterFS GlusterFS deb packages <deb.packages@gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+elif [ "$os" == "debian" ]; then
+	sed -i "1i glusterfs (${version}-1) ${flavor}; urgency=low\n\n * GlusterFS ${version} GA\n\n -- Gluster Packager <glusterpackager@download.gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
+
+    echo "Done"
+    exit
+fi
+
+# we are using the same builder machine to build so we are running the "pbuilder
+# create" everytime to create the chroot according to the os and flavor we want to build.
+echo "creating chroot for ${os} ${flavor}"
+sudo pbuilder create --distribution ${flavor} --mirror ${mirror} --debootstrapopts --keyring=/usr/share/keyrings/${os}-archive-keyring.gpg
+
+echo "Building glusterfs-${version} for ${os} ${flavor} using the chroot and .dsc we created"
+
+# have to use the .dsc file inside the ${os}${flavor} folder
+sudo pbuilder build ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc | tee build.log
+
+#move the packages to packages directory.
+mv /var/cache/pbuilder/result/glusterfs*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
```

Comment:
I am not sure jenkins used can erase file in /var/cache , so mv would fail.

Reply:
@kalebskeithley Any suggestions on this?

Reply:
I think it can safely replaced by "cp", at least.

Reply:
done

Reply:
Seems to still be mv :/

Reply:
But these files should be removed to avoid eventual filling up of space, How is such case handled generally with jenkins? @mscherer